### ISSUE TITLE
Unify effects to enchantments system

### ISSUE THREAD

**Issue by Maleclypse** [CleverRaven/Cataclysm-DDA#51307]
# Is your feature request related to a problem? Please describe.
Currently effects have a limited number of stats they can modify compared to the enchantment system.  However, they have a number of modifications they can make that can't be made in the enchantment system.  

# Describe the solution you'd like
Unify these two systems so that either can modify all the stats currently modifiable by either.

# Describe alternatives you've considered
There may be other ways of doing this that I'm missing.

# Additional context
Add any other context (such as mock-ups, proof of concepts or screenshots) about the feature request here.

**Comment by Maleclypse** [comment]
@MylieDaniels I wasn't sure if one of your reworks is impacting this down the line.  No stress if not, trying to make sure that if you PR something that closes this I remember to link it to this issue.

**Comment by MylieDaniels** [comment]
Yep, the long term plan is to have Traits, Enchants, and EOCs instead of Mutations, Bionics, Effects, EOCs, Enchants, et cetera
